### PR TITLE
Add `inferencedata_to_tidy_draws`

### PR DESCRIPTION
**Pull request by damonbayer**

### CODE REVIEW

File: R/inferencedata_df_to_tidy_draws.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#' Convert InferenceData DataFrame to nest tidy_draws
```
Comment (dylanhmorris) [inline R/inferencedata_df_to_tidy_draws.R:1]:
```suggestion
#' Convert InferenceData DataFrame to nested tibble of tidy_draws
```

File: R/inferencedata_df_to_tidy_draws.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#'
+#' @param idata InferenceData DataFrame (the result of calling arviz.InferenceData.to_dataframe in Python)
+#'
+#' @return A nested tibble, with columns group and data. Each element of data is a tidy_draws data frame
+#' @export
+#'
+#' @examples
+
```
Comment (dylanhmorris) [inline R/inferencedata_df_to_tidy_draws.R:9]:
```suggestion
```

File: R/inferencedata_df_to_tidy_draws.R [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+              string = x[2],
+              pattern = "(?<=\\[).+(?=\\])",
+              replacement = str_replace_all(x[3], "\\s", "_")
+            )
+            # white space not allowed in dimension names
+            x <- x[-3]
+          }
+          x
+        }) |>
+        map_chr(\(x) str_c(x, collapse = ","))
```
Comment (dylanhmorris) [inline R/inferencedata_df_to_tidy_draws.R:34]: This column name cleanup has some relatively opaque operations, maybe worth making a non-anonymous helper function (even an internal one) with some documentation and perhaps even inline comments?

File: R/inferencedata_df_to_tidy_draws.R [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      .after = .iteration
+    ) |>
+    pivot_longer(-starts_with("."),
+      names_sep = ",",
+      names_to = c("group", "name")
+    ) |>
+    group_by(group) |>
+    nest() |>
+    mutate(data = map(data, \(x) pivot_wider(x, names_from = name) |> tidy_draws()))
+}
```
Comment (dylanhmorris) [inline R/inferencedata_df_to_tidy_draws.R:48]: Either need to import the relevant functions here or qualify namespaces. In general I have been minimizing `@importFrom` in forecasttools

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
Thanks @damonbayer! A few minor things and one more major request: it would be nice to have a test for this function.

File: R/inferencedata_df_to_tidy_draws.R (hunk 4 above)
Comment (damonbayer) [reply R/inferencedata_df_to_tidy_draws.R]: addressed in https://github.com/CDCgov/forecasttools/pull/7/commits/c6a691e281132a9dde783ee29d2c303ec58422c2

**Review by dylanhmorris — APPROVED** [review]
LGTM, thanks @damonbayer

**Pull request merged**